src/libecalc/core/models/compressor/train/variable_speed_compressor_train_common_shaft_multiple_streams_and_pressures.py

```diff
@@ -405,6 +407,70 @@ def get_max_standard_rate_per_stream(
 
         return np.array(max_std_rate_per_stream)
 
+    def _get_max_rate_for_single_stream_single_timestep_brute_force(
+        self,
+        suction_pressure: float,
+        target_discharge_pressure: float,
+        rate_per_stream: NDArray[np.float64],
+        stream_to_maximize: int,
+    ) -> float:
+        """NB: Constraining to calculating maximum_rate for ingoing streams for now - the need is to figure out how much rate
+        can be added.
+
+        """
+        stream_to_maximize_connected_to_stage_no = self.streams[stream_to_maximize].connected_to_stage_no
+
+        # if it is not an ingoing stream --> currently no calculations done
+        # Fixme: what should be returned? 0.0, NaN or something else?
+        if not self.streams[stream_to_maximize].is_inlet_stream:
+            return 0.0
```

Comment:
raise error? If that happens a developer made a mistake?

Reply:
It's not an error. This is if we take something out (e.g. for export) between two compressors. I guess in reality we will be limited by what has entered the compressor train earlier - we could in theory take everything out - so it could be the sum of the rates for all the streams that has entered/left the compressor train at earlier stages.